### DISCUSSION TITLE
`fake_button_press` for X11 `Window`s

### DISCUSSION THREAD

**Discussion by ervinpopescu** [qtile/qtile#4701] [Ideas] [answered]
Feature request to implement this directly into `qtile`, in order to have this as a usecase:
```sh
for i in $(qtile cmd-obj -o cmd -f windows | qtile_to_json | jq '.[] | select(.name | contains("something")) | .id')
do
    echo $i
    qtile cmd-obj -o window $i -f fake_button_press
done
```
or its python equivalent.

**Comment by elParaguayo** [comment, marked as answer]
What are you wanting to achieve with this? `fake_button_press` just triggers the `button_click` handler which is useful for our tests but it doesn't actually send button presses to windows.

Could you use `xdotool`?

**Reply by ervinpopescu** [reply]
Well, let's say that I have a program that spawns a million windows (a firefox iCloud tab that spawns a file chooser for each file  I want to download, for example) instead of doing the obvious and opening just one for all files I want to download.

This could be solved by using some sort of iCloud to local filesystem integration, but `rclone`, my go to in these situations, does not support iCloud, and I didn't find any other useful one.

In this situation it seems obvious that I can get the coordinates of the `Save` button and just click that, but I was looking for something more robust and reusable.

LE: Maybe `xdotool` doesn't sound so bad, after all.
